### ISSUE TITLE
Fallback image / page for offline experience

### ISSUE THREAD

**Issue by pcvonz** [magento/pwa-studio#493]
## This issue is for the following packages:

[x] `venia-concept`
[ ] `pwa-buildpack`
[ ] `peregrine`
[ ] `pwa-devdocs`
[ ] `upward-js`
[ ] `upward-spec`

## This issue is a:

[ ] Bug
[x] Feature suggestion
[ ] Documentation issue
[ ] Other (Please Specify)

### Environment

| Question                    |    Answer     |
| --------------------------- | :-----------: |
| Magento version             |  |
| Operating System + version  |  |
| node.js version (`node -v`) |  |
| npm version (`npm -v`)      |  |

## Description

As a shopper of the Venia storefront, I want basic feedback when an image or page can't be displayed when offline

**Current experience:**

![offline_no_cache](https://user-images.githubusercontent.com/6378569/48360152-6c648080-e653-11e8-8ee9-ad190af19cf4.png)

![image](https://user-images.githubusercontent.com/6378569/48359979-fcee9100-e652-11e8-81e7-7b3b29053dfe.png)

## Expected result:

When offline and displaying a non cached image or web page, insert a fallback that tells the user the image/page isn't cached.

## Possible solutions:

Use Workbox to insert a fallback image. Reference the [Workbox docs](https://developers.google.com/web/tools/workbox/guides/advanced-recipes#

For the fallback page, the `NotFound` Root Component in pull request #486 will need to be restyled. 

## Dependencies

This pull request needs to be merged before we can work on this issue: https://github.com/magento-research/pwa-studio/pull/486

This issue also requires assets. Pinging @soumya-ashok.

**Comment by zetlen** [comment]
@pcvonz There are some irregularities with ServiceWorker caching right now, but now that we've merged your awesome work in #486, I don't know what the repro is for this issue. Is it still valid?

**Comment by ericerway** [comment]
Suggestion for @jimbo at grooming to consider content vs. asset fallback similar to what is being done for Categories on Homepage.

**Comment by JStein92** [comment]
Waiting for fallback image mockup from @soumya-ashok to proceed

**Comment by soumya-ashok** [comment]
Background color #f6f6f6 will cover the intended area of the image.
Icon in #707070 will be centered in the area. The icon will scale based on the area available. I will slack you the SVG @JStein92 since Github doesn't support that format here.

Specs for icon sizing based on image context. Not all cases are specifically covered, but should provide a reasonable guide.
![venia-placeholder-image](https://user-images.githubusercontent.com/25960021/49767291-836fb000-fc9d-11e8-96b1-0f7ea9cb7a1d.png)

**Comment by JStein92** [comment]
@ericerway @zetlen Can we spin off the Offline/Not Found page styling to a separate issue? I feel like although it's in the scope of 'offline functionality' it doesn't belong in the same issue as fallback images  Also it would be a good `first issue`/`help wanted`.

**Comment by sirugh** [comment]
I closed #672 for multiple reasons noted in the comment on the PR. I still think this issue may be valid though I cannot reproduce it. Basically we need a way to render a category page in offline mode with *some* of the data fetched, but currently the app renders a data fetch error.

[![Image from Gyazo](https://i.gyazo.com/b6ae7f41d6623a64b9f4cb16c34f2ade.png)](https://gyazo.com/b6ae7f41d6623a64b9f4cb16c34f2ade)

**Comment by awilcoxa** [comment]
did offline image get merged in previously referenced PR #486? @jimbo @zetlen

**Comment by awilcoxa** [comment]
Closing this legacy issue and will track as requirement in more recent issue: https://github.com/magento/pwa-studio/issues/1958
